kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/future/FutureMatcherTest.kt

```diff
@@ -2,17 +2,24 @@ package com.sksamuel.kotest.matchers.future
 
 import io.kotest.assertions.throwables.shouldThrowMessage
 import io.kotest.core.spec.style.StringSpec
-import io.kotest.matchers.future.*
-import kotlinx.coroutines.DelicateCoroutinesApi
+import io.kotest.matchers.future.shouldBeCancelled
+import io.kotest.matchers.future.shouldBeCompleted
+import io.kotest.matchers.future.shouldBeCompletedExceptionally
+import io.kotest.matchers.future.shouldCompleteExceptionallyWith
+import io.kotest.matchers.future.shouldNotBeCancelled
+import io.kotest.matchers.future.shouldNotBeCompleted
+import io.kotest.matchers.future.shouldNotBeCompletedExceptionally
+import io.kotest.matchers.future.shouldNotCompleteExceptionallyWith
+import kotlinx.coroutines.asCoroutineDispatcher
 import kotlinx.coroutines.delay
-import kotlinx.coroutines.newSingleThreadContext
 import kotlinx.coroutines.withContext
 import java.util.concurrent.CompletableFuture
+import java.util.concurrent.Executors
 
 class FutureMatcherTest : StringSpec({
+
    suspend fun runOnSeparateThread(block: () -> Unit) {
-      @OptIn(DelicateCoroutinesApi::class)
-      newSingleThreadContext("separate").use {
+      Executors.newSingleThreadExecutor().asCoroutineDispatcher().use {
```

Comment:
Why are we going back to explicit JDK executors?

Reply:
I just figured newSingleThreadContext has the warnings, the jvm code doesn't, and I thought they were the same basically. I'll switch it back, but curious if the kotlin one has advantages?

Reply:
Reverted

Reply:
> curious if the kotlin one has advantages?

I usually do some research when selecting such stuff and then quickly forget all the details. In this case, I'd say the dispatchers provided by kotlinx.coroutines
* are multiplatform,
* create named threads on the JVM (easier debugging),
* use a slightly different executor on the JVM (and my assumption is that the coroutines folks usually have sound reasons for their choices).